aquadoggo/src/domain.rs

```diff
@@ -202,44 +208,60 @@ pub async fn next_args<S: StorageProvider>(
 /// ## Compute and return next entry arguments
 pub async fn publish<S: StorageProvider>(
     store: &S,
-    entry_encoded: &EntrySigned,
-    operation_encoded: &OperationEncoded,
+    schema: &Schema,
+    encoded_entry: &EncodedEntry,
+    plain_operation: &PlainOperation,
+    encoded_operation: &EncodedOperation,
 ) -> Result<NextEntryArguments> {
-    ////////////////////////////////
-    // DECODE ENTRY AND OPERATION //
-    ////////////////////////////////
+    //////////////////
+    // DECODE ENTRY //
+    //////////////////
 
-    let entry = decode_entry(entry_encoded, Some(operation_encoded))?;
-    let operation = Operation::from(operation_encoded);
-    let author = entry_encoded.author();
+    let entry = decode_entry(encoded_entry)?;
+    let author = entry.public_key();
     let log_id = entry.log_id();
     let seq_num = entry.seq_num();
 
-    ///////////////////////////
-    // VALIDATE ENTRY VALUES //
-    ///////////////////////////
+    //////////////////////////////////
+    // VALIDATE ENTRY AND OPERATION //
+    //////////////////////////////////
+
+    // @TODO: Check this validation flow is still correct.
 
     // Verify that the claimed seq num matches the expected seq num for this author and log.
     let latest_entry = store.get_latest_entry(&author, log_id).await?;
     let latest_seq_num = latest_entry.as_ref().map(|entry| entry.seq_num());
-    is_next_seq_num(latest_seq_num.as_ref(), seq_num)?;
+    is_next_seq_num(latest_seq_num, seq_num)?;
```

Comment:
This step is not needed I don't think as `validate_operation_with_entry()` will catch the case where the encoded seq num doesn't match the expected one.

Reply:
Yes, we can remove it. The only thing we loose is a nicer error message.